routes.py

```
@@ -1,28 +1,34 @@
 from flask import Flask, redirect, request, url_for, render_template
-from app import db
-from models import User
-from app import app
-from models import Blogpost, User
 from flask_sqlalchemy import SQLAlchemy
 
+from app import app, db
+from models import Blogpost, User
 
 
-@app.route('/home')
+@app.route('/posts', methods=['GET', 'POST'])
 def home():
-    posts = Blogpost.query.order_by(Blogpost.posted_on.desc()).all()
-    return render_template('home.html', posts=posts)
-    
-@app.route('/posts/new', methods=['GET', 'POST'])
-def new_post():
     if request.method == 'POST':
-        post_title = request.form['title']
-        post_content = request.form['post']
-        post_author = request.form['author']
-
-        new_post = Blogpost(title=post_title, content=post_content, posted_by=post_author)
-
-        db.session.add(new_post)
-        db.session.commit()
+        create_post()
         return redirect('/posts')
+    elif request.Method == 'GET':
+        posts = get_posts()
+        return render_template('home.html', posts=posts)
     else:
-        return render_template('create_post.html')
+        # return unimplemented error page
```

You want to return a NotImplemented error code if a method that wasn't isn't implemented is called

---

Since there is no field for the author's name in the 'create_post' file, I think the author's name will be gotten from the db relationship when the author logs in, so is it still valid having the request.author field in the 'create_post'?